Dockerfile

```diff
@@ -1,25 +1,50 @@
-ARG PYTHON_IMAGE_TAG=3.13.4-slim-bullseye
+ARG PYTHON_IMAGE_TAG=3.14.0-slim-trixie
 
 FROM python:$PYTHON_IMAGE_TAG AS apt_base
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update
 
 # Add 3rd-party APT repositories.
 FROM apt_base AS apt_repos
-RUN apt-get install -y --no-install-recommends gnupg2 apt-transport-https software-properties-common curl
-RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
-RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+RUN apt-get install -y --no-install-recommends gnupg2 apt-transport-https curl
+RUN install -m 0755 -d /etc/apt/keyrings
+
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
+RUN chmod a+r /etc/apt/keyrings/docker.asc
+RUN echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
+  bookworm stable" | tee -a /etc/apt/sources.list.d/docker.list
+
 RUN echo 'deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main' | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
-RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg -o /usr/share/keyrings/cloud.google.gpg
+RUN chmod a+r /usr/share/keyrings/cloud.google.gpg
+
+RUN curl -fsSL https://packages.confluent.io/deb/8.0/archive.key | gpg --dearmor > /etc/apt/keyrings/confluent.gpg
+RUN chmod a+r /etc/apt/keyrings/confluent.gpg
+
+RUN echo "deb [signed-by=/etc/apt/keyrings/confluent.gpg] https://packages.confluent.io/deb/8.0 stable main" > /etc/apt/sources.list.d/confluent.list && \
+    echo "deb [signed-by=/etc/apt/keyrings/confluent.gpg] https://packages.confluent.io/clients/deb/ bookworm main" >> /etc/apt/sources.list.d/confluent.list
 
 # Download, build and install Python packages.
 FROM apt_base AS python_packages
+COPY --from=apt_repos /etc/apt/keyrings/confluent.gpg /etc/apt/keyrings/confluent.gpg
+COPY --from=apt_repos /etc/apt/sources.list.d/confluent.list /etc/apt/sources.list.d/confluent.list
 ENV PIP_NO_CACHE_DIR=1
 ENV UV_PROJECT_ENVIRONMENT="/usr/local/"
-RUN apt-get install -y --no-install-recommends build-essential cmake libssl-dev zlib1g-dev libffi-dev
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
+    build-essential \
+    cmake \
+    libssl-dev \
+    zlib1g-dev \
+    libffi-dev \
+    librdkafka-dev \
+    libev4 \
+    libev-dev
```

Comment:
I think these can be removed if we upgrade to python-driver 3.29.5 https://pypi.org/project/scylla-driver/3.29.5, which seems to have wheel already

Reply:
Those gonna stay, for next upgrade of python.